src/Subscriber.ts

```diff
@@ -111,7 +128,22 @@ class SafeSubscriber<T> extends Subscriber<T> {
     if (!this.isStopped) {
       if (this._complete) {
         this._complete();
+        this._reportMessageError(true);
+      } else {
+        this.unsubscribe();
+      }
+    }
+  }
+
+  private _reportMessageError(isStopped: boolean): void {
+    const { parent, syncErrorThrown } = this;
+    if (parent) {
```

Comment:
I think we could wrap this whole block in an `if (syncErrorThrown)` since it's innards don't do anything unless that is `true`.